lib/datasets.go

```diff
@@ -400,39 +443,68 @@ func (r *DatasetRequests) Remove(p *RemoveParams, numDeleted *int) error {
 
 // LookupParams defines parameters for looking up the body of a dataset
 type LookupParams struct {
-	Format        dataset.DataFormat
+	Format        string
 	FormatConfig  dataset.FormatConfig
-	Path          string
+	PathString    string
+	Ref           *repo.DatasetRef
+	Selector      string
+	Concise       bool
 	Limit, Offset int
 	All           bool
 }
 
 // LookupResult combines data with it's hashed path
 type LookupResult struct {
-	Path string `json:"path"`
-	// TODO: Rename to Body
-	Data []byte `json:"data"`
+	Path  string          `json:"path"`
+	Data  dataset.Dataset `json:"data"`
+	Bytes []byte          `json:"bytes"`
 }
 
 // LookupBody retrieves the dataset body
-func (r *DatasetRequests) LookupBody(p *LookupParams, data *LookupResult) (err error) {
+func (r *DatasetRequests) LookupBody(p *LookupParams, res *LookupResult) (err error) {
 	if r.cli != nil {
-		return r.cli.Call("DatasetRequests.LookupBody", p, data)
+		return r.cli.Call("DatasetRequests.LookupBody", p, res)
 	}
 
 	if p.Limit < 0 || p.Offset < 0 {
 		return fmt.Errorf("invalid limit / offset settings")
 	}
 
-	bodyPath, bufData, err := actions.LookupBody(r.node, p.Path, p.Format, p.FormatConfig, p.Limit, p.Offset, p.All)
+	if p.Ref == nil {
+		ref, err := repo.ParseDatasetRef(p.PathString)
+		if err != nil {
+			return err
+		}
+		p.Ref = &ref
+	}
+
+	if p.Ref.Dataset == nil {
+		ds, err := dsfs.LoadDataset(r.node.Repo.Store(), p.PathString)
+		if err != nil {
+			return fmt.Errorf("error loading dataset")
+		}
+		res.Data = *ds
+	} else {
+		err = res.Data.Decode(p.Ref.Dataset)
+		if err != nil {
+			return err
+		}
+	}
+
+	df, err := dataset.ParseDataFormatString(p.Format)
 	if err != nil {
 		return err
 	}
 
-	*data = LookupResult{
-		Path: bodyPath,
-		Data: bufData,
+	bodyPath, bufData, err := actions.LookupBody(r.node, &res.Data, df, p.FormatConfig, p.Limit, p.Offset, p.All)
+	if err != nil {
+		return err
 	}
+
+	// TOOD: --format flag
```

Comment:
halp my find & replace doesn't know TOOD 😛

Reply:
This TOOD / TODO is out of date anyway, removed.

Reply:
even better!